src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs

```diff
@@ -29,8 +36,13 @@ public class AnonymousParameterTypes : TypedRule<SwaggerParameter>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// An entity fails this rule if it has a schema, and that schema is an anonymous type
         /// </summary>
-        public override Category Severity => Category.Info;
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public override bool IsValid(SwaggerParameter entity) =>
+            entity?.Schema == null || AnonymousTypesRule.IsValid(entity.Schema);
```

Comment:
@dsgouda if Schema is null then isValid method should return true, that's why I'm checking for it here.